internal/jujuapi/httpproxy_integration_test.go

```diff
@@ -0,0 +1,207 @@
+	c.Assert(err, gc.IsNil)
+	model := &dbmodel.Model{UUID: sql.NullString{String: env.Models[0].UUID, Valid: true}}
+	err = s.JIMM.Database.GetModel(ctx, model)
+	c.Assert(err, gc.IsNil)
+	s.model = model
+	err = s.JIMM.GetCredentialStore().PutControllerCredentials(ctx, model.Controller.Name, "user", "psw")
+	c.Assert(err, gc.IsNil)
+}
+
+func (s *httpProxySuite) TestCharmHTTPServe(c *gc.C) {
+	ctx := context.Background()
+	httpProxier := jujuapi.HTTPProxier{JIMM: s.JIMM}
+	expectU, expectP, err := s.JIMM.GetCredentialStore().GetControllerCredentials(ctx, s.model.Controller.Name)
+	c.Assert(err, gc.IsNil)
+	// we expect the controller to respond with TLS
+	fakeController := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		u, p, _ := r.BasicAuth()
+		c.Assert(u, gc.Equals, names.NewUserTag(expectU).String())
+		c.Assert(p, gc.Equals, expectP)
+		_, err = w.Write([]byte("OK"))
+		c.Assert(err, gc.IsNil)
+	}))
+	defer fakeController.Close()
+	controller := s.model.Controller
+	pemData := pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: fakeController.Certificate().Raw,
+	})
+	controller.CACertificate = string(pemData)
+
+	tests := []struct {
+		description    string
+		setup          func()
+		url            string
+		modelUUID      string
+		statusExpected int
+	}{
+		{
+			description: "good",
+			setup: func() {
+				newURL, _ := url.Parse(fakeController.URL)
+				controller.PublicAddress = newURL.Host
+				err = s.JIMM.Database.UpdateController(ctx, &controller)
+				c.Assert(err, gc.IsNil)
+			},
+			url:            fmt.Sprintf("/model/%s/charms", s.model.UUID.String),
+			modelUUID:      s.model.UUID.String,
+			statusExpected: http.StatusOK,
+		},
+		{
+			description: "model not existing",
+			setup: func() {
+			},
+			url:            fmt.Sprintf("/model/%s/charms", "54d9f921-c45a-4825-8253-74e7edc28066"),
+			modelUUID:      "54d9f921-c45a-4825-8253-74e7edc28066",
+			statusExpected: http.StatusNotFound,
+		},
+	}
+
+	for _, test := range tests {
+		test.setup()
+		req, err := http.NewRequest("POST", test.url, nil)
+		c.Assert(err, gc.IsNil)
+		recorder := httptest.NewRecorder()
+		rctx := chi.NewRouteContext()
+		rctx.URLParams.Add("uuid", test.modelUUID)
+		ctx := context.WithValue(req.Context(), chi.RouteCtxKey, rctx)
+		httpProxier.ServeHTTP(ctx, recorder, req.WithContext(ctx))
+		resp := recorder.Result()
+		defer resp.Body.Close()
+		c.Assert(resp.StatusCode, gc.Equals, test.statusExpected)
+	}
+}
+
+type e2eProxySuite struct {
+	websocketSuite
+}
+
+var _ = gc.Suite(&e2eProxySuite{})
+
+func (s *e2eProxySuite) TestLocalCharmDeploy(c *gc.C) {
+	conn := s.open(c, &api.Info{
+		ModelTag:  s.Model.ResourceTag(),
+		SkipLogin: false,
+	}, s.AdminUser.Name)
+
+	client, err := charms.NewLocalCharmClient(conn)
+	c.Assert(err, gc.IsNil)
+	charmArchive := testcharms.Repo.CharmArchive(c.MkDir(), "dummy")
```

Comment:
If possible, please don't use the word "dummy", because it's against the inclusive language.

Reply:
it's the default local charm in juju tests. If I use another name it doesn't work because it is not found.

[search in juju code](https://github.com/search?q=repo%3Ajuju%2Fjuju+testcharms.Repo.CharmArchive%28c.MkDir%28%29%2C+%22dummy%22%29&type=code)